gdal/frmts/wcs/wcsutils.cpp

```diff
@@ -0,0 +1,698 @@
+/******************************************************************************
+ *
+ * Project:  WCS Client Driver
+ * Purpose:  Implementation of utilities.
+ * Author:   Frank Warmerdam, warmerdam@pobox.com
+ *
+ ******************************************************************************
+ * Copyright (c) 2006, Frank Warmerdam
+ * Copyright (c) 2008-2013, Even Rouault <even dot rouault at mines-paris dot org>
+ * Copyright (c) 2017, Ari Jolma
+ * Copyright (c) 2017, Finnish Environment Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#include "ogr_spatialref.h"
+#include "wcsutils.h"
+#include <algorithm>
+
+#define DIGITS "0123456789"
+
+void Swap(double &a, double &b)
+{
+    double tmp = a;
+    a = b;
+    b = tmp;
+}
+
+CPLString URLEncode(CPLString str)
+{
+    char *pszEncoded = CPLEscapeString(str, -1, CPLES_URL );
+    CPLString str2 = pszEncoded;
+    CPLFree( pszEncoded );
+    return str2;
+}
+
+CPLString URLRemoveKey(const char *url, CPLString key)
+{
+    CPLString retval = url;
+    key += "=";
+    while (true) {
+        size_t pos = retval.ifind(key);
+        if (pos != std::string::npos) {
+            size_t end = retval.find("&", pos);
+            retval.erase(pos, end - pos + 1);
+        } else {
+            break;
+        }
+    }
+    if (retval.back() == '&') {
+        retval.pop_back();
+    }
+    return retval;
+}
+
+std::vector<CPLString> SwapFirstTwo(std::vector<CPLString> array)
```

Comment:
Either use
std::vector<CPLString>& SwapFirstTwo(std::vector<CPLString>& array) if you want modify in place and return a reference to the modified object
Or
std::vector<CPLString> SwapFirstTwo(const std::vector<CPLString>& array) if you plan to return a modified copy